std/array.asm

```diff
@@ -6,6 +6,15 @@ let<T> len: T[] -> int = [];
 /// Evaluates to the array [f(0), f(1), ..., f(length - 1)].
 let<T> new: int, (int -> T) -> T[] = |length, f| std::utils::fold(length, f, [], |acc, e| (acc + [e]));
 
+/// Returns a new array equal to arr except that the element at index i is x.
+let<T> set_element: T[], int, T -> T[] = |arr, i, x| {
+    let _ = std::check::assert(i < len(arr), || "Index out of bounds");
+    map_enumerated(arr, |j, y| if i == j { x } else { y })
+};
+
+/// Returns a new array of length l containing the elements of arr starting at index start.
+let<T> sub_array: T[], int, int -> T[] = |arr, start, l| std::array::new(l, |i| arr[start + i]);
```

Comment:
We don't need a bound check, because `arr[start + i]` would panic if `start + l > len(arr)`, right?

Reply:
Yes